src/app/utils/history/add-object-change.ts

```diff
@@ -0,0 +1,34 @@
+import { Manuscript } from 'app/types/manuscript';
+import { get, set } from 'lodash';
+import { Change } from 'app/utils/history/change';
+import { cloneManuscript } from 'app/utils/state.utils';
+
+export class AddObjectChange<T> extends Change {
+  constructor(private path: string, private object: T, private idField: string) {
+    super();
+  }
+
+  get isEmpty(): boolean {
+    return false;
+  }
+
+  applyChange(manuscript: Manuscript): Manuscript {
+    const originalSection = get(manuscript, this.path);
+
+    if (!Array.isArray(originalSection)) {
+      throw new TypeError('Trying to make AddObject change on a non-array section');
```

Comment:
Does anything catch these?

Reply:
no. I put them there for debugging purposes so that we get an error if we make a mistake in the path parameter and it points to a non array part of a manuscript